redfish-core/lib/pcie_slots.hpp

```diff
@@ -230,12 +242,166 @@ inline void handlePCIeSlotCollectionGet(
     });
 }
 
+// Get all valid  PCIe Slots which are on the given chassis
+inline void getValidPCIeSlotList(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisID, const std::string& chassisPath,
+    std::function<void(
+        const std::vector<std::pair<std::string, std::string>>&)>&& callback)
+{
+    BMCWEB_LOG_DEBUG("Get properties for PCIeSlots associated to chassis = {}",
+                     chassisID);
+
+    // get PCIeSlots that are in chassis
+    constexpr std::array<std::string_view, 1> pcieSlotIntf{
+        "xyz.openbmc_project.Inventory.Item.PCIeSlot"};
+
+    dbus::utility::getAssociatedSubTree(
+        chassisPath + "/inventory",
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"), 0,
+        pcieSlotIntf,
+        [asyncResp, chassisID, chassisPath, callback{std::move(callback)}](
+            const boost::system::error_code& ec,
+            const dbus::utility::MapperGetSubTreeResponse& subtree) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("D-Bus response error on GetSubTree {}",
```

Comment:
Is check for EBADR needed here?

Reply:
I think as long as /xyz/openbmc_project/inventory exists it would be return an empty resource collection if "xyz.openbmc_project.Inventory.Item.PCIeSlot" doesn't exists

Reply:
The subtree emptiness is checked in the next statement.
```
        if (subtree.empty())
```
So, the current code will be kept as is